src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -576,6 +589,25 @@ public static function get_merged_data( $origin = 'custom' ) {
 			_deprecated_argument( __FUNCTION__, '5.9.0' );
 		}
 
+		if ( ! in_array( $origin, WP_Theme_JSON::VALID_ORIGINS, true ) ) {
+			$origin = 'custom';
+		}
+
+		// Map origins to block cache values.
+		$cache_map = array(
```

Comment:
what's the advantage of declaring this map over just using the origin?

Reply:
Oh, I see. The existing calls to `has_same_registered_blocks` do not use the `origin`. It's clear if they do, that'd be a little nice change before this PR (or as part of it).

Reply:
I agree it's unfortunate that the names don't match, but I also think we shouldn't make a change to that at this point due to backward compatibility risk. If we want to update them, that should be in its own decoupled issue due to the additional considerations.